### PR TITLE
Prepare for 0.7 release

### PR DESCRIPTION
**Pull request by ludwigschwardt**
More than I thought...

Ignoring the massive but cosmetic flake8 changes, the only other changes involve test cleanups. I silence the rcond FutureWarning that polluted the test output, as well as a custom condition warning (but I also check that the latter happens as expected). I also had to relax one test to make it pass on NumPy 1.15.0.

### CODE REVIEW

File: scikits/fitting/__init__.py [hunk 1]
```diff
@@ -68,7 +70,7 @@
 Helper functions
 ----------------
 
-- :func:`squash` : Flatten array, but not necessarily all the way to a 1-D array
+- :func:`squash` : Flatten array, but not necessarily all the way to 1-D array
```
Comment (bmerry) [inline scikits/fitting/__init__.py]: I preferred the way this was.

File: scikits/fitting/rbf.py [hunk 2]
```diff
@@ -32,7 +32,7 @@
 
 
 class RbfScatterFit(ScatterFit):
-    """Do radial basis function (RBF) interpolation of scattered multi-dimensional data.
+    """Do radial basis function (RBF) interpolation of scattered multi-dim data.
```
Comment (bmerry) [inline scikits/fitting/rbf.py]: I preferred the original.

File: scikits/fitting/tests/test_linlstsq.py [hunk 3]
```diff
@@ -76,7 +76,7 @@ def test_vs_numpy(self):
         """LinearLeastSquaresFit: Compare fitter to np.linalg.lstsq."""
         interp = LinearLeastSquaresFit()
         interp.fit(self.x, self.y)
-        params = np.linalg.lstsq(self.x.T, self.y)[0]
+        params = np.linalg.lstsq(self.x.T, self.y, rcond=None)[0]
```
Comment (bmerry) [inline scikits/fitting/tests/test_linlstsq.py]: This unfortunately gives an error (`TypeError: a float is required`) in numpy < 0.14. So if you do this, you'll need to specify a minimum of 0.14 in setup.py.

File: NEWS.rst [hunk 4]
```diff
@@ -1,6 +1,12 @@
 History
 =======
 
+0.7 (2018-09-20)
+----------------
+
+* Python 3 support
+* Clean up tests and more flake8 (line lengths)
```
Comment (bmerry) [inline NEWS.rst:8]: Add note about fixes for newer numpy?

**Review by bmerry — CHANGES_REQUESTED** [review]

File: scikits/fitting/__init__.py (hunk 1 above)
Comment (ludwigschwardt) [reply scikits/fitting/__init__.py]: Fixed.

File: scikits/fitting/rbf.py (hunk 2 above)
Comment (ludwigschwardt) [reply scikits/fitting/rbf.py]: Fixed.

File: scikits/fitting/tests/test_linlstsq.py (hunk 3 above)
Comment (ludwigschwardt) [reply scikits/fitting/tests/test_linlstsq.py]: I'll do `rcond=-1` instead, which should be backwards compatible. This only matters for one test anyway.

File: NEWS.rst (hunk 4 above)
Comment (ludwigschwardt) [reply NEWS.rst:8]: Since it only affects one test, it is not that special.

**Review by bmerry — APPROVED** [review]

**Pull request merged**